frontend/class-opengraph.php

```diff
@@ -649,4 +558,208 @@ public function publish_date() {
+	 */
+	private function set_images() {
+		if ( is_front_page() ) {
+			$this->get_front_page_image();
+		}
+
+		if ( is_singular() ) {
+			$this->get_singular_image();
+		}
+
+		$this->get_default_image();
+	}
+
+	/**
+	 * If the frontpage image exists, call add_image
+	 */
+	private function get_front_page_image() {
+		if ( $this->options['og_frontpage_image'] !== '' ) {
+			$this->add_image( $this->options['og_frontpage_image'] );
+		}
+	}
+
+	/**
+	 * Get the images of the singular post.
+	 */
+	private function get_singular_image() {
+		global $post;
+
+		if ( $this->get_opengraph_image() ) {
+			return;
+		}
+
+		if ( $this->get_featured_image( $post->ID ) ) {
+			return;
+		}
+
+		$this->get_content_images( $post );
+	}
+
+	/**
+	 * Get default image and call add_image
+	 */
+	private function get_default_image() {
+		if ( count( $this->images ) == 0 && isset( $this->options['og_default_image'] ) && $this->options['og_default_image'] !== '' ) {
+			$this->add_image( $this->options['og_default_image'] );
+		}
+	}
+
+	/**
+	 * If opengraph-image is set, call add_image and return true
+	 *
+	 * @return bool
+	 */
+	private function get_opengraph_image() {
+		$ogimg = WPSEO_Meta::get_value( 'opengraph-image' );
+		if ( $ogimg !== '' ) {
+			$this->add_image( $ogimg );
+
+			return true;
+		}
+	}
+
+	/**
+	 * If there is a featured image, check image size. If image size is correct, call add_image and return true
+	 *
+	 * @param $post_id
+	 *
+	 * @return bool
+	 */
+	private function get_featured_image( $post_id ) {
+		if ( function_exists( 'has_post_thumbnail' ) && has_post_thumbnail( $post_id ) ) {
+			/**
+			 * Filter: 'wpseo_opengraph_image_size' - Allow changing the image size used for OpenGraph sharing
+			 *
+			 * @api string $unsigned Size string
+			 */
+			$thumb = wp_get_attachment_image_src( get_post_thumbnail_id( $post_id ), apply_filters( 'wpseo_opengraph_image_size', 'original' ) );
+
+			if ( $this->check_image_size( $thumb ) ) {
+				$this->add_image( $thumb[0] );
+
+				return true;
+			}
+		}
+	}
+
+	/**
+	 * Filter: 'wpseo_pre_analysis_post_content' - Allow filtering the content before analysis
+	 *
+	 * @api string $post_content The Post content string
+	 *
+	 * @param object $post The post object.
+	 */
+	private function get_content_images( $post ) {
+		$content = apply_filters( 'wpseo_pre_analysis_post_content', $post->post_content, $post );
+
+		if ( preg_match_all( '`<img [^>]+>`', $content, $matches ) ) {
+			foreach ( $matches[0] as $img ) {
+				if ( preg_match( '`src=(["\'])(.*?)\1`', $img, $match ) ) {
+					$this->add_image( $match[2] );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Check size of image. If image is too small, return false, else return true
+	 *
+	 * @param array $img
+	 *
+	 * @return bool
+	 */
+	private function check_image_size( $img_data ) {
+		//Get the width and height of the image.
+		if ( $img_data[1] < 200 || $img_data[2] < 200 ) {
```

Comment:
Yep, see: http://codex.wordpress.org/Function_Reference/wp_get_attachment_image_src
check_image_size is called by get_featured_image with a thumb. The thumb is a wp_get_attachment_image_src, that always returns an ordered array. This function should not be called if it's not to check the featured image size. I'll edit the function name to make this more clear.